pkg/metrics/controller_test.go

```diff
@@ -141,22 +168,31 @@ func TestControllerGameServersTotal(t *testing.T) {
 	generateGsEvents(16, agonesv1.GameServerStateStarting, "", c.gsWatch)
 	generateGsEvents(1, agonesv1.GameServerStateUnhealthy, "", c.gsWatch)
 
-	c.sync()
+	expected := 96
+	assert.Eventually(t, func() bool {
+		list, err := c.gameServerLister.GameServers(gs.ObjectMeta.Namespace).List(labels.Everything())
+		require.NoError(t, err)
+		return len(list) == expected
+	}, 5*time.Second, time.Second)
+	// write a good message if something goes wrong
```

Comment:
Not sure I can follow this comment. Could you please remove or explain more what is it for?

Reply:
Updated comment, PTAL!

Reply:
Now it is much more verbose, I like it.